File: sentimental-cash/cash.py
from cs50 import get_float

while True:
    cash = get_float("change: ")
    cents = round(cash * 100)
    if cents >= 1:
        break

quarters = 0
while cents >= 25:
    quarters += 1
    cents -= 25

dimes = 0
while cents >= 10:
    dimes += 1
    cents -= 10

nickels = 0
while cents >= 5:
    nickels += 1
    cents -= 5

pennies = 0
while cents >= 1:
    pennies += 1
    cents -= 1


print(quarters + dimes + nickels + pennies)

File: cash/cash.c
// least amount of coins for change owed
// coins available - 25c, 10c, 5c, 1c.
// input - change owed
// output - number of coins

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 1);

    int quarters = 0;
    while (cents >= 25)
    {
        quarters++;
        cents -= 25;
    }
    int dimes = 0;
    while (cents >= 10)
    {
        dimes++;
        cents -= 10;
    }
    int nickels = 0;
    while (cents >= 5)
    {
        nickels++;
        cents -= 5;
    }
    int pennies = 0;
    while (cents >= 1)
    {
        pennies++;
        cents -= 1;
    }

    int coins;
    {
        coins = quarters + dimes + nickels + pennies;
        printf("%i\n", coins);
    }
}
